homeassistant/components/rova/sensor.py

```diff
@@ -64,39 +75,71 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
-    add_entities: AddEntitiesCallback,
+    async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Create the Rova data service and sensors."""
-
-    zip_code = config[CONF_ZIP_CODE]
-    house_number = config[CONF_HOUSE_NUMBER]
-    house_number_suffix = config[CONF_HOUSE_NUMBER_SUFFIX]
-    platform_name = config[CONF_NAME]
-
-    # Create new Rova object to  retrieve data
-    api = Rova(zip_code, house_number, house_number_suffix)
-
-    try:
-        if not api.is_rova_area():
-            LOGGER.error("ROVA does not collect garbage in this area")
-            return
-    except (ConnectTimeout, HTTPError):
-        LOGGER.error("Could not retrieve details from ROVA API")
-        return
+    """Set up the rova sensor platform through yaml configuration."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=config,
+    )
+    if (
+        result["type"] == FlowResultType.CREATE_ENTRY
+        or result["reason"] == "already_configured"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version="2024.7.0",
```

Comment:
```suggestion
            breaks_in_ha_version="2024.10.0",
```
Deprecation period is 6 months, let's aim for it to go in this release